src/wallet/scriptpubkeyman.cpp

```diff
@@ -1836,6 +1836,10 @@ void DescriptorScriptPubKeyMan::MarkUnusedAddresses(const CScript& script)
 void DescriptorScriptPubKeyMan::AddDescriptorKey(const CKey& key, const CPubKey &pubkey)
 {
     LOCK(cs_desc_man);
+    if (m_map_keys.find(pubkey.GetID()) != m_map_keys.end()) {
```

Comment:
I think it would be better to have this inside of `AddDescriptorKeyWithDB`. Additionally, it needs to check `m_map_crypted_keys` otherwise this won't work when the wallet is encrypted.

Reply:
Done